be/src/exec/vectorized/json_scanner.cpp

```diff
@@ -80,13 +80,13 @@ StatusOr<ChunkPtr> JsonScanner::get_next() {
     } catch (simdjson::simdjson_error& e) {
         auto err_msg = "Unrecognized json format, stop json loader.";
         LOG(WARNING) << err_msg;
-        return Status::DataQualityError(err_msg);
+        status = Status::DataQualityError(err_msg);
     }
     if (!status.ok()) {
         if (status.is_end_of_file()) {
             _cur_file_eof = true;
         } else {
-            return status;
+            return FileScanner::UnrecoverableError(status);
```

Comment:
why not return Status with UNRECOVERABLE_ERROR code. And origin status can be assigned to status error_message

Reply:
This is to return message to the FE to distinguish error that should pause the routine load. The error code is removed in `Status StreamLoadExecutor::rollback_txn(StreamLoadContext* ctx)`, so the error info in added in the error message rather than the error code.
https://github.com/StarRocks/starrocks/blob/0463dbd79322a7bcbc8106e1bd9b46b84721af05/be/src/runtime/stream_load/stream_load_executor.cpp#L221

Reply:
I see and thanks for you explanation